infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/Geocoding.kt

```diff
@@ -183,6 +183,11 @@ data class GeocodingContext(
     fun getAddress(coordinate: IPoint, decimals: Int = DEFAULT_TRACK_METER_DECIMALS): Pair<TrackMeter, IntersectType>? =
         getDistance(coordinate)?.let { (dist, type) -> getAddress(dist, decimals) to type }
 
+    fun getDistanceAndAddress(coordinate: IPoint, decimals: Int = DEFAULT_TRACK_METER_DECIMALS): Triple<Double, TrackMeter, IntersectType>? =
+        getDistance(coordinate)?.let { (dist, type) ->
+            Triple(dist, getAddress(dist, decimals), type)
+        }
```

Comment:
En sinällään kauheasti tykkää `Triple`:stä, mutta tähän se tuntui luontevalta. Tätä käytetään toistaiseksi vaan yhdessä paikassa (ja sielläkin destructuroiden), niin en viitsinyt lähteä tekemään uutta dataclassiakaan. Toisaalta tuota `IntersectType`:ä ei kyllä tarvita siellä toistaiseksi ainoassa kutsupäässä, mutta potentiaalisissa muissa käyttökohteissa se voisi ehkä kuitenkin olla fiksua informaatiota, joten jätin sen sinne.

Reply:
Joo onhan nämä kaikki tuplat ja triplat vähän huonoja. Sinällään jos sen haluaa vaihtaa niin myös dataclassit voi destructuroida joten se ei sinällään estä. 

Jos tuo paluutyyppi olisi olio, se voisi palvella kaikki käyttötavat (m-arvon antaminen paluuarvossa osoitteen mukana olisi ihan luonnollista) eikä tarvitsisi toista funkkaria lainkaan. Se taisi itse asiassa jossain kohtaa olla tuolla paluuarvossa mukana mutta poistui kun sitä ei enää tarvittu.

Reply:
Meillä on aika monta paikkaa joissa tarvitaan M-arvoja, mutta osoitteita ei, joten sinällään näiden funktioiden yhdistäminen kokonaan yhdeksi funktioksi ei välttämättä ole järkevää. Sellaisia käyttöpaikkoja joissa tarvitaan joko molempia tai osoitetta (jonka laskemiseen taas tarvitaan M-arvoa) löytyi jokunen, ja voisin vaihtaa ne käyttämään tätä.

Reply:
Näin tarkemmalla silmäyksellä oikeastaan voisin jättää myös tuon `getAddress`:in tuonne sellaisenaan. Meillä on todella monta paikkaa, jossa M-arvo ei ole ollenkaan kiinnostava asia, ja ne kaikki tarvisi refaktoroida niiden käyttöpaikkojen näkökulmasta huonompaan suuntaan